File: Block Party-Objective_C/Block Party-Objective_C/Entities/Hero.m
//
//  Hero.m
//  Block Party-Objective_C
//
//  Created by Qi Feng Huang on 10/31/15.
//  Copyright © 2015 hackathon. All rights reserved.
//

#import "Hero.h"
#import "Constants.h"

@implementation Hero

-(instancetype)initWithTexture:(SKTexture *)texture{
    //sprite has a clear color over it and takes on the size of the texture
    self = [super initWithTexture:texture color: [UIColor clearColor] size:texture.size];
    
    self.name = @"hero";
    self.zPosition = 75;
    
    //create physics body
    SKPhysicsBody* physicsBody = [SKPhysicsBody bodyWithRectangleOfSize:self.size];
    physicsBody.usesPreciseCollisionDetection = true;
    
    //collision won't cause hero to move slower
    physicsBody.restitution = 0;
    physicsBody.friction = 0;
    
    //Set physics category to HERO
    physicsBody.categoryBitMask = HERO;
    
    //Hero collisdes with Boundary; can't move past boundary
    physicsBody.collisionBitMask = BOUNDARY;
    
    //Can come into contact with charm and enemy
    physicsBody.contactTestBitMask = ENEMY | CHARM;
    self.physicsBody = physicsBody;
    
    return self;
}


@end

File: Block Party/Hero.swift
//
//  Dot.swift
//  dot
//
//  Created by Qi Feng Huang on 1/3/15.
//  Copyright (c) 2015 Qi Feng Huang. All rights reserved.
//


import SpriteKit


class Hero : SKSpriteNode{
    
    //rec is the playable scene
    init(spriteTexture: SKTexture){
        
        super.init(texture: spriteTexture, color: UIColor.clearColor(), size: spriteTexture.size())
        name = "hero"
        zPosition = 75
        
        //create physics body
        let physicsbody = SKPhysicsBody(rectangleOfSize: self.size)
        physicsbody.usesPreciseCollisionDetection = true
        
        physicsbody.restitution = 0
        physicsbody.friction = 0
        physicsbody.categoryBitMask = PhysicsCategory.Hero
        physicsbody.collisionBitMask = PhysicsCategory.Boundary
        physicsbody.contactTestBitMask = PhysicsCategory.Enemy | PhysicsCategory.Charm
        self.physicsBody = physicsbody
    }
    
    required init(coder aDecoder : NSCoder){
        fatalError("NSCoding not supported")
    }
    

}
